scripts/apps/monitoring/directives/WidgetGroup.js

```diff
@@ -0,0 +1,364 @@
+    $location, $anchorScroll, activityService, $rootScope, gettextCatalog, datetime, metadata) {
+    const services = {
+        search: search,
+        api: api,
+        superdesk: superdesk,
+        desks: desks,
+        cards: cards,
+        $timeout: $timeout,
+        $q: $q,
+        $location: $location,
+        $anchorScroll: $anchorScroll,
+        activityService: activityService,
+        $rootScope: $rootScope,
+        gettextCatalog: gettextCatalog,
+        datetime: datetime,
+        metadata: metadata
+    };
+
+    return {
+        scope: {
+            stage: '=',
+            total: '=',
+            allowed: '=',
+            showEmpty: '=?',
+            maxItems: '=?',
+            selected: '=?',
+            action: '&',
+            filter: '='
+        },
+        link: function(scope, elem) {
+            var criteria;
+
+            scope.page = 1;
+            scope.fetching = false;
+            scope.items = [];
+
+            /**
+              * Generates Identifier to be used by track by expression.
+              */
+            scope.generateTrackByIdentifier = function(item) {
+                return search.generateTrackByIdentifier(item);
+            };
+
+            scope.preview = function(item) {
+                superdesk.intent('preview', 'item', item);
+            };
+
+            scope.edit = function(item) {
+                if (item._type === 'ingest') {
+                    var activity = superdesk.findActivities({action: 'list', type: 'ingest'}, item)[0];
+
+                    activityService.start(activity, {data: {item: item}}).then((item) => {
+                        initEdit(item);
+                    });
+                } else {
+                    initEdit(item);
+                }
+
+                function initEdit(item) {
+                    superdesk.intent('edit', 'item', item).then(null, () => {
+                        superdesk.intent('view', 'item', item);
+                    });
+                }
+            };
+
+            function getProvider(criteria) {
+                var provider = 'archive';
+
+                if (scope.stage.type && (desks.isOutputType(scope.stage.type) || scope.stage.type === 'search')) {
+                    provider = 'search';
+                }
+
+                if (criteria.repo && criteria.repo.indexOf(',') === -1) {
+                    provider = criteria.repo;
+                }
+
+                return provider;
+            }
+
+            function queryItems(queryString) {
+                var oldSize;
+
+                criteria = cards.criteria(scope.stage, queryString);
+                if (!scope.fetching) {
+                    scope.setLoading(true);
+                }
+
+                oldSize = criteria.source.size;
+                criteria.source.from = 0;
+
+                if (scope.page > 0 && criteria.source) {
+                    criteria.source.size = (scope.page - 1) * oldSize;
```

Comment:
Looks that the same solution is used on monitoring: 
https://github.com/superdesk/superdesk-client-core/blob/master/scripts/apps/monitoring/directives/MonitoringGroup.js#L364
If you don't load all items it will appear issues if an item is deleted from list (ex: moved to other stage or publish).
I think for server is not much difference on returning 100 or 1000 rows, and usually a user don't scroll more than 40 pages.
Due to unique keys on items React is fast on rendering so I think that all old vs new list processing on search.updateItems is not needed anymore for items from monitoring groups.